From c646afab7bed3d8548669756f837396ecf6e9775 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Wed, 15 Mar 2017 16:34:45 +0200
Subject: [PATCH 1581/5242] MLK-14454: video: mipi_dsi_northwest: Fix leaking
 a reset_control

commit  4022b22bbdc09ac1ec650c7eb5bb5d2599ce2ec4 from
https://source.codeaurora.org/external/imx/linux-imx.git

The mipi_dsi_disp_init function takes a reference to a reset_control
during initialization and does not release it.

This becomes a problem on 4.9 because reset_control_get has been split
into exclusive mode(default) and shared(which need to be marked).
Leaking a reference counts looks like attempting to fetch a second
reference to a controller and this WARNs and returns -EBUSY.

Fix by releasing it at the end of the function.

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mipi_dsi_northwest.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/mipi_dsi_northwest.c b/drivers/video/fbdev/mxc/mipi_dsi_northwest.c
index fb30d0f..a955ba44 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi_northwest.c
+++ b/drivers/video/fbdev/mxc/mipi_dsi_northwest.c
@@ -165,11 +165,13 @@ static int mipi_dsi_disp_init(struct mxc_dispdrv_handle *disp,
 	ret = mipi_dsi_lcd_init(mipi_dsi, setting);
 	if (ret) {
 		dev_err(dev, "failed to init mipi dsi lcd\n");
-		return ret;
+		goto out;
 	}
 
 	dev_info(dev, "MIPI DSI dispdv inited\n");
 
+out:
+	reset_control_put(reset);
 	return ret;
 }
 
-- 
1.7.9.5

